src/fights.ts

```diff
@@ -1737,6 +1737,59 @@ const freeRunFightSources = [
       },
     }
   ),
+  // Try for an ultra-rare with mayfly runs ;)
+  new FreeRunFight(
+    () =>
+      have($item`mayfly bait necklace`) &&
+      canAdventure($location`Cobb's Knob Menagerie, Level 1`) &&
+      // Max combat length is 4 rounds, Basic elementals have 40 HP base
+      4 * maxPassiveDamage() < 39 + numericModifier("Monster Level") &&
```

Comment:
down side here is if you do have some passive damage and could still free run but not do all the rounds this will stop it from free running with mayfly bait

Reply:
Is the best way to do that just to make another copy of this FreeRunFight where if we do have passive damage, we only do the free runs and no pickpocketing?

The other option is I make extra checks for each step whether the monsters HP is high enough, but that would require monster manuel